### PR TITLE
Android/update versions gradle

### PR DESCRIPTION
**Pull request by shoatman**
- Upgrade mockito android version to latest
- Updated compile sdk version
- Updated target sdk version
- Added java 8 compat (https://developer.android.com/studio/write/java8-support)

### CODE REVIEW

File: common/build.gradle [hunk 1]
```diff
@@ -57,8 +67,8 @@ android {
             minifyEnabled false
             debuggable false
             testCoverageEnabled true
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-            consumerProguardFiles 'consumer-rules.pro'
+            //proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+            //consumerProguardFiles 'consumer-rules.pro'
```
Comment (AdamBJohnsonx) [inline common/build.gradle:71]: Why comment this out?

Comment (shoatman) [reply common/build.gradle:71]: Good catch.  That was an investigation and turned out to be a red herring.

File: common/build.gradle [hunk 2]
```diff
@@ -18,7 +18,16 @@ def bothTestType = "BothTest"
...
+
+    compileOptions {
+        // Flag to enable support for the new language APIs
+        coreLibraryDesugaringEnabled true
+        // Sets Java compatibility to Java 8
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    compileSdkVersion 30//rootProject.ext.compileSdkVersion
```
Comment (AdamBJohnsonx) [inline common/build.gradle]: should this still be commented out?  Probably remove the comment or attach a work item to fix whatever is wrong with the versions file.

File: common/build.gradle (hunk 1 above)
Comment (shoatman) [reply common/build.gradle]: fixed

File: common/build.gradle (hunk 2 above)
Comment (shoatman) [reply common/build.gradle]: Nope.  I fixed.

File: gradle.properties [hunk 3]
```diff
@@ -15,4 +15,4 @@ org.gradle.jvmargs=-Xmx4g -XX:MaxPermSize=2048m -Dkotlin.daemon.jvm.options\="-X
 versionName=1.5.9
 
 # For OneAuth default abiSelection
-abiSelection=x86
\ No newline at end of file
+abiSelection=x86_64
```
Comment (AdamBJohnsonx) [inline gradle.properties:18]:
```suggestion
abiSelection=x86_64

```

File: common/build.gradle [hunk 4]
```diff
@@ -18,6 +18,15 @@ def bothTestType = "BothTest"
 project.ext.vstsUsername = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME") != null ? System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME") : project.findProperty("vstsUsername")
 project.ext.vstsPassword = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN") != null ? System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN") : project.findProperty("vstsMavenAccessToken")
 android {
+
+    compileOptions {
+        // Flag to enable support for the new language APIs
+        coreLibraryDesugaringEnabled true
```
Comment (AdamBJohnsonx) [inline common/build.gradle:24]: Yay!

File: uiautomationutilities/build.gradle [hunk 5]
```diff
@@ -10,6 +10,15 @@ project.ext.vstsPassword = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN
...
+    compileOptions {
+        // Flag to enable support for the new language APIs
+        coreLibraryDesugaringEnabled true
+        // Sets Java compatibility to Java 8
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
     compileSdkVersion rootProject.ext.compileSdkVersion
     buildToolsVersion rootProject.ext.buildToolsVersion
```
Comment (AdamBJohnsonx) [inline uiautomationutilities/build.gradle:23]: Any way we can axe this, since it's going to get overridden by the new versions of Gradle anyway?

**Review by AdamBJohnsonx — APPROVED** [review]
Minor comments only.

File: uiautomationutilities/build.gradle (hunk 5 above)
Comment (shoatman) [reply uiautomationutilities/build.gradle:23]: I don't see why not.

File: common/build.gradle [hunk 6]
```diff
@@ -5,9 +5,12 @@ apply plugin: 'maven-publish'
 apply from: 'versioning/version_tasks.gradle'
 apply plugin: 'jacoco'
 
-// To include Robolectric tests in the Jacoco report, flag -> "includeNolocationClasses" is set to true
+//To include Robolectric tests in the Jacoco report, flag -> "includeNolocationClasses" is set to true
 tasks.withType(Test) {
     jacoco.includeNoLocationClasses = true
+    //This needs to be excluded for JDK 11
+    //SEE: https://support.circleci.com/hc/en-us/articles/360047926852-Android-Builds-Fail-with-java-lang-ClassNotFoundException-jdk-internal-reflect-GeneratedSerializationConstructorAccessor1-
+    jacoco.excludes = ['jdk.internal.*']
```
Comment (shahzaibj) [inline common/build.gradle:13]: interesting

File: common/build.gradle [hunk 7]
```diff
@@ -103,6 +120,9 @@ android {
 }
 
 dependencies {
+    //Java 8 - Desugaring
+    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.0.9'
```
Comment (shahzaibj) [inline common/build.gradle:124]: The version can go in the version file

File: common/src/androidTest/AndroidManifest.xml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        </intent>
+        <!-- Required for API Level 30 to make sure we can detect browsers that support custom tabs -->
+        <!-- https://developers.google.com/web/updates/2020/07/custom-tabs-android-11#detecting_browsers_that_support_custom_tabs -->
+        <intent>
+            <action android:name=
+                "android.support.customtabs.action.CustomTabsService" />
+        </intent>
+    </queries>
+
+</manifest>
```
Comment (shahzaibj) [inline common/src/androidTest/AndroidManifest.xml:35]: new line needed

File: common/src/androidTest/AndroidManifest.xml [hunk 9]
```diff
@@ -0,0 +1,35 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (shahzaibj) [inline common/src/androidTest/AndroidManifest.xml:1]: Can you add a note why this was needed? Because as far as I know the one in common `main` dir should be being used already

File: labapi/build.gradle [hunk 10]
```diff
@@ -50,6 +50,10 @@ publishing {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'io.swagger:swagger-annotations:1.5.17'
+    implementation group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.0'
+    implementation group: 'com.sun.xml.bind', name: 'jaxb-core', version: '2.3.0'
+    implementation group: 'com.sun.xml.bind', name: 'jaxb-impl', version: '2.3.0'
+    implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
```
Comment (shahzaibj) [inline labapi/build.gradle:56]: Why are these needed?

File: uiautomationutilities/build.gradle [hunk 11]
```diff
@@ -122,6 +132,9 @@ project.afterEvaluate{
 }
 
 dependencies {
+    //Java 8 - Desugaring
+    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.0.9'
```
Comment (shahzaibj) [inline uiautomationutilities/build.gradle:136]: The version can go in version file

File: testutils/build.gradle [hunk 12]
```diff
@@ -123,6 +132,10 @@ project.afterEvaluate{
 }
 
 dependencies {
+
+    //Java 8 - Desugaring
+    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.0.9'
```
Comment (shahzaibj) [inline testutils/build.gradle:137]: The version can go in version file

**Review by shahzaibj — COMMENTED** [review]
Left some comments

File: common/build.gradle [hunk 13]
```diff
@@ -17,7 +20,19 @@ def bothTestType = "BothTest"
 
 project.ext.vstsUsername = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME") != null ? System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME") : project.findProperty("vstsUsername")
 project.ext.vstsPassword = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN") != null ? System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN") : project.findProperty("vstsMavenAccessToken")
+
 android {
+
+
```
Comment (AdamBJohnsonx) [inline common/build.gradle:26]: nit: extra lines

File: labapi/build.gradle (hunk 10 above)
Comment (AdamBJohnsonx) [reply labapi/build.gradle:56]: They've been removed from java 11's default packaging.

File: common/build.gradle (hunk 7 above)
Comment (shoatman) [reply common/build.gradle:130]: Done.

File: common/src/androidTest/AndroidManifest.xml (hunk 9 above)
Comment (shoatman) [reply common/src/androidTest/AndroidManifest.xml:1]: I would if I could remember.  I'm going to leave for now since it doesn't have anything in it.

File: common/src/androidTest/AndroidManifest.xml (hunk 8 above)
Comment (shoatman) [reply common/src/androidTest/AndroidManifest.xml:35]: I didn't understand this one.  Am resolving for now please re-open if necessary.
